content/en/security/application_security/enabling/compatibility/java.md

```diff
@@ -6,45 +6,131 @@ type: multi-code-lang
 code_lang_weight: 0
 ---
 
+## ASM capabilities support
+
+The following ASM capabilities are supported in the Java library, for the specified tracer version:
+
+| ASM capability                   | Minimum Java tracer version |
+| -------------------------------- | ----------------------------|
+| Threat Detection | 1.8.0  |
+| Threat Protection| 1.9.0 |
+| Open Source Vulnerability detection | 1.1.4 |
+| Custom Code Vulnerability detection | 1.15.0|
+
+The minimum tracer version to get all supported ASM capabilities for Java is 1.15.0.
+
+**Note**: Threat Protection requires enabling [Remote Configuration][2], which is included in the listed minimum tracer version.      
+
 ## Language and framework compatibility
 
 ### Supported Java versions
+The Java Tracer supports automatic instrumentation for the following Oracle JDK and OpenJDK JVM runtimes.
 
-The Datadog library supports Java JRE 1.8 and higher of both Oracle JDK and OpenJDK, on the following architectures:
-- Linux (GNU) x86, x86-64
-- Alpine Linux (musl) x86, x86-64
-- macOS (Darwin) x86, x86-64
-- Windows (msvc) x86, x86-64
+| JVM versions | Operating Systems                                                               | Support level                       | Tracer version |
+| -------------| ------------------------------------------------------------------------------- | ----------------------------------- | -------------- |
+| 18 to 19     | Windows (x86, x86-64)<br>Linux (x86, x86-64, arm64)<br>Mac (x86, x86-64, arm64) | [Beta](#levels-of-support)               | Latest         |
+| 8 to 17      | Windows (x86, x86-64)<br>Linux (x86, x86-64)<br>Mac (x86, x86-64)               | [GA](#levels-of-support)                   | Latest         |
+| 8 to 17      | Linux (arm64)<br>Mac (arm64)                                                    | [Beta](#levels-of-support)               | Latest         |
+| 7            | Windows (x86, x86-64)<br>Linux (x86, x86-64)<br>Mac (x86, x86-64)               | [Maintenance](#levels-of-support) | v0             |
+| 7            | Linux (arm64)<br>Mac (arm64)                                                    | [End-of-life](#levels-of-support)         | v0             |
 
 Datadog does not officially support any early-access versions of Java.
 
 You can monitor application security for Java apps running in Docker, Kubernetes, AWS ECS, and AWS Fargate.
 
-### Supported frameworks
+## Integrations
+
+### Levels of framework support
```

Comment:
@sourabhkatti I moved this table from where it was to just below Integrations as it looked to be in the wrong place